File: cpp/n507_PerfectNum.cpp
// https://leetcode-cn.com/problems/perfect-number/
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
#include <math.h>
using namespace std;

class Solution {
public:
    bool checkPerfectNumber(int num) {
        if (num <= 1) return false;
        int res = 1;
        for (int i = 2; i < int(sqrt(num))+1; ++i){
            if (num % i == 0) res = res + i + num/i;
        }
        cout << res;
        return (num == res);
    }
};

int main(){
    return 0;
}

File: python/n507_PerfectNum.py
# https://leetcode-cn.com/problems/perfect-number/
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num <= 1: return False
        res = 1
        for i in range(2, int(math.sqrt(num))+1):
            if num % i == 0: res = res + i + num/i
        return num == res
        
  
if __name__ == "__main__":
    pass